docs/source/dev/test.rst

```diff
@@ -0,0 +1,107 @@
+        ${_TARGET_NAME}
+        ${_FILES_SOURCE})
+    target_link_libraries(
+        ${_TARGET_NAME}
+        PRIVATE common)
+
+    set_target_properties(${_TARGET_NAME} PROPERTIES FOLDER "test/unit")
+
+    add_test(NAME ${_TARGET_NAME} COMMAND ${_TARGET_NAME} ${_ALPAKA_TEST_OPTIONS})
+
+where the first line sets the name of the test that ``CMake`` will build for execution. Furthermore, it is necessary to add
+the folder in which the test is located to the ``CMakeLists.txt`` file under the ``test/unit`` path:
+
+.. code-block::
+    add_subdirectory("dev/")
+
+to make sure that ``CMake`` will build the test written in the ``.cpp`` file under that directory.
+
+After these steps, it all comes to writing the *actual* test. First of all, the necessary header libraries should be
+included the ``.cpp`` file:
+
+.. code-block:: c++
+    #include <alpaka/dev/Traits.hpp>
+    #include <alpaka/test/acc/TestAccs.hpp>
+    #include <catch2/catch.hpp>
+
+The first ``include`` statement targets specifically the header in which the functionality is included; with the second one
+only accelerators corresponding to enabled backends are selected for testing, while the third one is necessary to integrate
+Catch2.
+
+One or more test cases are then introduced with the ``TEMPLATE_LIST_TEST_CASE`` macro. It takes three arguments:
+* a free form test name (must be unique)
+* a tag
+* the accelerator(s) that must run the test case(s) (i.e. ``alpaka::test::TestAccs`` targets all the accelereators selected
+by the ``TestAccs`` header).
+
+Some aliases might be useful:
+
+.. code-block:: c++
+    using Dev = alpaka::Dev<TestType>;
+    using Pltf = alpaka::Pltf<Dev>;
+
+The template parameter for ``alpaka::Dev`` is ``TestType``, which basically target the specific accelerator that corresponds
+to the backend which is executing the test.
+
+Since the function that must be tested has a device object as argument, it is necessary to define one:
+
+.. code-block:: c++
+    Dev const dev(alpaka::getDevByIdx<Pltf>(0u));
+
+Having all the necessary objects well defined, the function can be called and the result stored in an object:
+
+.. code-block:: c++
+    auto const warpExtent = alpaka::getWarpSize(dev);
+
+Finally, the condition under which the test succeed should be defined with the ``REQUIRE`` macro:
+
+.. code-block:: c++
+    REQUIRE(warpExtent > 0);
+
+Such condition depends on what must be actually tested (i.e. the size of the warp can never be equal or less than 0).
+Furthermore, other additional conditions can be set as well:
+
+.. code-block:: c++
+    REQUIRE(std::is_same_v<decltype(warpExtent), int>);
+
+which checks that the type of the warp size is integer (cannot be float).
+
+.. hint::
+    An alternative to the ``REQUIRE`` macro is the ``STATIC_REQUIRE`` macro. The difference between them is that if the
+    test fails, in the first case the failure will be notified at run time, while in the second case it will result in
+    a compile time error.
```

Comment:
I would omit this and point to the Catch2 documentation. There is so much more, like checking for exceptions, handling floating points etc.